storage/zynq7000-sdcard/sdstorage_srv.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Phoenix-RTOS
+ *
+ * SD Card libstorage-based server
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Jacek Maksymowicz
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/file.h>
+#include <sys/threads.h>
+#include <posix/utils.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <cache.h>
+#include <storage/storage.h>
+#include <libjffs2.h>
+#include <libext2.h>
+
+#include "sdstorage_dev.h"
+
+#define LOG_TAG "sdstorage_srv"
+/* clang-format off */
+#define LOG_ERROR(str, ...) do { fprintf(stderr, LOG_TAG " error: " str "\n", ##__VA_ARGS__); } while (0)
+#define TRACE(str, ...)     do { if (0) fprintf(stderr, LOG_TAG " trace: " str "\n", ##__VA_ARGS__); } while (0)
+/* clang-format on */
+
+
+static ssize_t storage_read(id_t id, offs_t offs, void *buff, size_t len)
+{
+	ssize_t res = -ENOSYS;
+	storage_t *strg = storage_get(GET_STORAGE_ID(id));
+
+	if (strg == NULL || strg->dev == NULL || (offs + len) > strg->size || buff == NULL) {
```

Comment:
MISRA, applies in other places too
```suggestion
	if ((strg == NULL) || (strg->dev == NULL) || ((offs + len) > strg->size) || (buff == NULL)) {
```